### PR TITLE
Feat: handle deeplink in runtime

### PR DESCRIPTION
**Pull request by NickKhalow**
# Pull Request Description

### Prerequisites
- [ ] List any required setup steps
- [ ] Include environment/configuration requirements

### Test Steps
1. First step
2. Second step
3. Expected result after step 2
4. ...

### Additional Testing Notes
- Note any edge cases to verify
- Mention specific areas that need careful testing
- List known limitations or potential issues

## Quality Checklist
- [ ] Changes have been tested locally
- [ ] Documentation has been updated (if required)
- [ ] Performance impact has been considered
- [ ] For SDK features: Test scene is included

## Code Review Reference
Please review our [Code Review Standards](https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md) before submitting.

### CODE REVIEW

File: Explorer/Assets/DCL/RuntimeDeepLink/DeepLinkSentinel.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                string content = await File.ReadAllTextAsync(DEEP_LINK_BRIDGE_PATH, token)!;
+
+                // Notify emitter that file has been consumed
+                File.Delete(DEEP_LINK_BRIDGE_PATH);
+
+                DeepLinkDTO dto = JsonUtility.FromJson<DeepLinkDTO>(content);
+                string? raw = dto.deeplink;
+                DeepLinkCreateResult deepLinkCreateResult = DeepLink.FromRaw(raw);
+
+                deepLinkCreateResult.Match((handle, raw),
```
[the remark this answers is not in the document]
Comment (NickKhalow) [reply Explorer/Assets/DCL/RuntimeDeepLink/DeepLinkSentinel.cs:58]: I'm not agree in terms of complexity, please clarify:
_You are wrapping the controller by the handle_
_You are wrapping the controller by the handle_
What controller you are referring to?

_It's not clear why error processing can't be done immediately and you need an enum for that_
Where would you suggest to put error processing? `DeepLink.FromRaw(raw)` method could complete or fail - result type suits the best here. If `FromRaw` will also handle printing it's not obvious what to return and why. I see benefits of result approach in its explicitness

[the remark this answers is not in the document]
Comment (mikhail-dcl) [reply Explorer/Assets/DCL/RuntimeDeepLink/DeepLinkSentinel.cs:64]: The current handling of deep links has a high cognitive complexity:
- It's not linear; it contains many hops and lambdas.
- It's an unconventional design for C# development. Developers working on the project should grasp the code with **minimal effort**, having no experience with Rust.
- If you bring unconventional patterns from other languages to the project it should be justified: e.g. with Media Urls it looks justified because:
   - There, the Union indeed contains different structures
   - Here the Union contains one structure and several errors definitions which can be represented via a single error string without hops

Here how I modifed it https://github.com/decentraland/unity-explorer/commit/43a9a3988764e3249b12fa78ef363ac02a63f83e:

- I got rid of all renums
- I made the flow as linear and conventional as possible
- I handle a deep link in a single controller, the error is produced by a single controller
- As a side effect of removing a special `Null` value I had to delete a playground but:
   - I believe it should be Unit Tested instead
   - You don't even need an abstraction; you simply create a file (or feed the content) and expect the input
   - It will be much beneficial to include such a test in the pipeline than checking manually in the playground (the latter is the only justification for `Null` implementation)
